<launch>
  <param name="/use_sim_time" value="true"/>

  <node
      pkg="map_server"
      type="map_server"
      name="map_server"
      args="$(find navigation_stage)/stage_config/maps/willow-full-0.05.pgm 0.05"
      respawn="false"
      />

  <node
      pkg="stage_ros"
      type="stageros"
      name="stageros"
      args="$(find shrink_inflation_recovery)/worlds/willow-pr2-5cm.world"
      respawn="false"
      >
    <param name="base_watchdog_timeout" value="0.2"/>
  </node>

  <node pkg="move_base" type="move_base" respawn="false" name="move_base_node" output="screen">
    <param name="footprint_padding" value="0.01" />
    <param name="controller_frequency" value="10.0" />
    <param name="controller_patience" value="3.0" />
    <param name="oscillation_timeout" value="30.0" />
    <param name="oscillation_distance" value="0.5" />

    <rosparam file="$(find navigation_stage)/move_base_config/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find navigation_stage)/move_base_config/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find navigation_stage)/move_base_config/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find navigation_stage)/move_base_config/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find navigation_stage)/move_base_config/base_local_planner_params.yaml" command="load" />

    <rosparam>
      recovery_behavior_enabled: true
      recovery_behaviors:
        - name: "shrink_inflation_local0"
          type: "shrink_inflation_recovery/ShrinkInflationRecovery"
        - name: "shrink_inflation_global0"
          type: "shrink_inflation_recovery/ShrinkInflationRecovery"
        - name: "conservative_reset"
          type: "clear_costmap_recovery/ClearCostmapRecovery"
        - name: "shrink_inflation_local1"
          type: "shrink_inflation_recovery/ShrinkInflationRecovery"
        - name: "shrink_inflation_global1"
          type: "shrink_inflation_recovery/ShrinkInflationRecovery"
        - name: "aggressive_reset"
          type: "clear_costmap_recovery/ClearCostmapRecovery"
        - name: "all_reset"
          type: "clear_costmap_recovery/ClearCostmapRecovery"
      shrink_inflation_local0:
        parameter_name: "/move_base_node/local_costmap/inflation_layer"
        inflation_radius: 0.1
      shrink_inflation_global0:
        parameter_name: "/move_base_node/global_costmap/inflation_layer"
        inflation_radius: 0.1
      conservative_reset:
        reset_distance: 2.0
      shrink_inflation_local1:
        parameter_name: "/move_base_node/local_costmap/inflation_layer"
        inflation_radius: 0.05
      shrink_inflation_global1:
        parameter_name: "/move_base_node/global_costmap/inflation_layer"
        inflation_radius: 0.05
      aggressive_reset:
        reset_distance: 1.0 # 0.5
      all_reset:
        reset_distance: 0.0
    </rosparam>
  </node>

  <include file="$(find navigation_stage)/move_base_config/amcl_node.xml"/>  

  <node
      name="rviz"
      pkg="rviz"
      type="rviz"
      args="-d $(find shrink_inflation_recovery)/config/single_robot.rviz"
      />

</launch>
